crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1236,6 +1224,57 @@ const fn is_python_whitespace(c: char) -> bool {
     )
 }
 
+enum LexedText<'a> {
+    Source { source: &'a str, range: TextRange },
+    Owned(String),
+}
+
+impl<'a> LexedText<'a> {
+    fn new(start: TextSize, source: &'a str) -> Self {
+        Self::Source {
+            range: TextRange::empty(start),
+            source,
+        }
+    }
+
+    fn push_matching(&mut self, c: char) {
```

Comment:
nit: I wonder if we could rename both `push_matching` and `push_different` as otherwise I found it difficult to understand what it does without looking at the implementation.

Reply:
Renaming is easy. The question is what better names are?

Reply:
I wonder if `push_different` is necessary with my other comment above as it's only used here. As such I can't think of anything right now 😅